File: tiq-e/strings/7/main.py
from time import perf_counter


class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if needle == '':
            return 0
        l = len(needle)
        for i in range(len(haystack) - len(needle) + 1):
            if haystack[i:i + l] == needle:
                return i
        return -1


haystack = "hello"
needle = "ll"

start = perf_counter()
result = Solution().strStr(haystack, needle)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/strings/7/main.rs
use std::time::Instant;

struct Solution;

impl Solution {
    pub fn str_str(haystack: String, needle: String) -> i32 {
        let window_size = needle.len();
        let needle = needle.as_bytes();
        if window_size == 0 {
            return 0i32;
        }
        match haystack
            .as_bytes()
            .windows(window_size)
            .position(|slice| slice == needle)
        {
            Some(x) => x as i32,
            _ => -1i32,
        }
    }
}

fn main() {
    let haystack: String = "hello".to_string();
    let needle: String = "ll".to_string();

    let start = Instant::now();
    let result = Solution::str_str(haystack, needle);
    let end = start.elapsed().as_micros();
    println!("Duration: {:?}", end);
    println!("Result: {:?}", result);
}
